examples/tutorials/core_api_pytorch_mnist/model_def_checkpoints.py

```diff
@@ -97,11 +100,23 @@ def test(args, model, device, test_loader, epoch, core_context, steps_completed)
     )
 
 
-# NEW: Define load_state function for restarting model training from existing checkpoint.
-def load_state(checkpoint_directory):
+# NEW: Define load_state function for restarting model training from existing checkpoint. Returns (.pt, int).
+# Also update load_state header to take trial info object as an argument.
+def load_state(checkpoint_directory, info):
     checkpoint_directory = pathlib.Path(checkpoint_directory)
-    with checkpoint_directory.joinpath("checkpoint.pt").open("rb") as f:
-        return torch.load(f)
+
+    with checkpoint_directory.joinpath("state").open("r") as f:
+        epochs_completed, ckpt_trial_id = [int(field) for field in f.read().split(",")]
+
+    # NEW: If the saved trial ID is the same as our current trial ID, we are resuming the trial, so we should return the model weights
+    # and the number of epochs already completed to resume from.
+    if ckpt_trial_id == info.trial.trial_id:
+        with checkpoint_directory.joinpath("checkpoint.pt").open("rb") as saved_model:
+            return torch.load(saved_model), epochs_completed
+    else:
+        # This is a new trial; load the model weight but return the number of epochs completed as 0.
+        with checkpoint_directory.joinpath("checkpoint.pt").open("rb") as saved_model:
+            return torch.load(saved_model), 0
```

Comment:
There's no need to duplicate your loading code in both if statements like this.

Why not:
```python

with checkpoint_directory.joinpath(...) as f:
    model = torch.load(saved_model)

# If trial ID does not match our current trial ID, we'll ignore epochs
# completed and start training from epoch_idx = 0
if ckpt_trial_id == current_trial_id:
    epochs_completed = 0

return model, epochs_completed
```